src/test/scala/mesosphere/marathon/health/MarathonHealthCheckManagerTest.scala

```diff
@@ -306,38 +305,5 @@ class MarathonHealthCheckManagerTest extends MarathonSpec with Logging {
     assert(hcManager.list(otherAppId) == otherHealthChecks)
   }
 
-  /**
-    * Captures the events send to the EventStream while the block is executing.
-    *
-    * For issue #2314 not only the end state is important. It is also important, which health checks
-    * were removed incorrectly and then readded afterwards.
-    */
-  private[this] def captureEvents(block: => Unit): Seq[MarathonEvent] = {
-    // yes, this is ugly. Since we only access it in the actor until it terminates, we do have
-    // the correct thread sync boundaries in place.
-
-    var capture = Vector.empty[MarathonEvent]
-    val captureEventsActor = actor {
-      new Act {
-        become {
-          case captureMe: MarathonEvent => capture :+= captureMe
-        }
-      }
-    }
-    eventStream.subscribe(captureEventsActor, classOf[MarathonEvent])
-    eventStream.subscribe(captureEventsActor, classOf[String])
-
-    try {
-      block
-    }
-    finally {
-      eventStream.unsubscribe(captureEventsActor)
-      captureEventsActor ! PoisonPill
-      val probe = TestProbe()
-      probe.watch(captureEventsActor)
-      probe.expectMsgClass(classOf[Terminated])
-    }
-
-    capture
-  }
+  def captureEvents = CaptureEvents(eventStream)
```

Comment:
Not sure I like this application of a case class. There is actually no pattern matching etc required or done, and imho the type of the def should be a function type – currently, it's implicitly referencing the apply method and its type is `CaptureEvents` while it should be `block: => Unit`.

Event though something like
`eventCaptor.capture {…}`
looks less elegant than your solution, I'd prefer this is production code. I can live with that in test code, just wanted to mention it :palm_tree: